plugins/source/plugin.go

```diff
@@ -159,7 +169,14 @@ func (p *Plugin) Sync(ctx context.Context, spec specs.Source, res chan<- *schema
 		return fmt.Errorf("failed to create execution client for source plugin %s: %w", p.name, err)
 	}
 	startTime := time.Now()
-	p.syncDfs(ctx, spec, c, tables, res)
+	switch spec.Scheduler {
+	case SchedulerDFS:
+		p.syncDfs(ctx, spec, c, tables, res)
+	case SchedulerRoundRobin:
+		p.syncRoundRobin(ctx, spec, c, tables, res)
+	default:
+		return fmt.Errorf("unknown scheduler %s. Options are: %v", spec.Scheduler, allSchedulers)
```

Comment:
default to dfs

Reply:
The default is handled elsewhere (in SetDefaults). We want the user to know if they've entered an invalid scheduler.